### PR TITLE
Adding 'card' print method

### LINKED ISSUE #93: Write a print method for the cards data frames
because many of the columns are list columns, the tibble print method makes it difficult to see the contents. the print method can make it easier to visually inspect the results.

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
Adding `print.card()` method.

@bzkrouse what do you think of the print method? Once we finalize, I'll finish the PR which will lead to some changes in snapshot testing I think.

image

What the print method does:
- converts  object to data frame so the list columns show the values (unlike a tibble print)
- prints between 10 and 20 rows
- messaging for the number of rows/columns that are not shown
- Removes warning and error columns if nothing to report
- if there are more than 6 columns, statistic_fmt_fn/contex columns removed
- A max of 9 characters are displayed for columns 'group##_level', 'variable_level', 'stat_label', and 'context'
- If a statistic is a double, then it is rounded to 3 decimal places
- If a proper function is found in the statistic_fmt_fn column, it's printed as "\<fn\>" to save space

**Reference GitHub issue associated with pull request.** _e.g., 'closes #1'_
closes #93

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] Ensure all package dependencies are installed: `devtools::install_dev_deps()`
- [ ] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [ ] If a bug was fixed, a unit test was added.
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# cards (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

**Comment by ddsjoberg** [comment]
I think this one is ready for review now, @bzkrouse ! Thanks!

File: README.md [hunk 1]
```diff
@@ -48,82 +48,77 @@ ARD Examples
...
+ard_categorical(ADSL, by = "ARM", variables = c("AGEGR1", "SEX"))
+#> {cards} data frame: 36 x 11
+#>    group1 group1_level variable variable_level stat_name stat_label statistic
+#> 1     ARM      Placebo   AGEGR1            <65         n          n        14
+#> 2     ARM      Placebo   AGEGR1            <65         p          %     0.163
+#> 3     ARM      Placebo   AGEGR1            >80         n          n        30
+#> 4     ARM      Placebo   AGEGR1            >80         p          %     0.349
+#> 5     ARM      Placebo   AGEGR1          65-80         n          n        42
+#> 6     ARM      Placebo   AGEGR1          65-80         p          %     0.488
+#> 7     ARM      Placebo   AGEGR1                        N          N        86
```
Comment (bzkrouse) [inline README.md:77]: I think the readme needs re-rendering after the last update :)

**Review by bzkrouse — APPROVED** [review]
This is really nice!!

File: README.md (hunk 1 above)
Comment (ddsjoberg) [reply README.md:77]: ahhh good catch! thank you!

**Pull request merged**